app/models/businesscustomer/ReviewDetails.scala

```diff
@@ -61,8 +61,13 @@ object ReviewDetails {
 
   implicit val writes = Json.writes[ReviewDetails]
 
-  implicit def convert(addr: BusinessMatchingAddress): Address =
-    Address(addr.line_1, addr.line_2, addr.line_3, addr.line_4, addr.postcode, Country("", addr.country))
+  implicit def convert(addr: BusinessMatchingAddress): Address = {
+      val country = models.countries collectFirst {
+        case country@Country(_, code) if code == addr.country => country
+      } getOrElse Country("", addr.country)
+
+      Address(addr.line_1, addr.line_2, addr.line_3, addr.line_4, addr.postcode, country)
+    }
```

Comment:
this change was necessary because I've noticed that previous implementation was creating empty country objects in some instances => so this conversion wasn't all the time right. For example in BusinessMatchingAddress there is country as String (place for country code) => so original conversion for United Kingdom was creating: Country("", "GB") but it should be: Country("United Kingdom", "GB").

Reply:
This sort of thing could have caused down stream issues. Good spot. 👍